### PR TITLE
Add is operator support to entities and component

### PR DESCRIPTION
**Pull request by adongare**
Ticket: https://oncprojectracking.healthit.gov/support/browse/BONNIEMAT-1158

- migrated nyc/jasmine to jest to address vulnerabilities
- removed ruby 2.6 and 2.7 from CI as they reached eol 
- removed nodejs 12 & 14 from CI as they reached eol
- regenerated models with "is" operator support
- regenerated index and browser js

Pull requests into cqm-models require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [ ] This pull request describes why these changes were made.
- [ ] Internal ticket for this PR:
- [ ] Internal ticket links to this PR
- [ ] Code diff has been done and been reviewed
- [ ] Tests are included and test edge cases
- [ ] Tests have been run locally and pass
- [ ] If applicable, the library version number in `package.json` and `cqm-models.gemspec` has been updated
- [ ] Cqm-execution fixtures have been updated with the update_cqm_execution_fixtures.sh script inside server-scripts using this branch in the cqm-converter

**Bonnie Reviewer:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Cypress Reviewer:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: app/assets/javascripts/QDMPatient.js [hunk 1]
```diff
@@ -103,15 +103,7 @@ QDMPatientSchema.methods.getByProfile = function getByProfile(profile, isNegated
...
-  const results = this.dataElements.filter(element => (element._type === `QDM::${profile}` || element._type === profile) && (isNegated === null || !!element.negationRationale === isNegated));
-  return results.map((result) => {
-    const removedMongooseItems = new AllDataElements[profile](result).toObject({ virtuals: true });
-    // toObject() will remove all mongoose functions but also remove the schema methods, so we add them back
-    Object.entries(result.schema.methods).forEach(([method_name, method]) => {
-      removedMongooseItems[method_name] = method;
-    });
-    return removedMongooseItems;
-  });
+  return this.dataElements.filter(element => (element._type === `QDM::${profile}` || element._type === profile) && (isNegated === null || !!element.negationRationale === isNegated));
```
Comment (adongare) [inline app/assets/javascripts/QDMPatient.js:106]: I don't know why we need to convert Mongoose objects to objects. and then add back methods to the converted object. toObject removed methods from nexted objects as well. if we have to add back methods from nested objects we have to iterate over all the props of the data element and add all the missing methods. I do not think we need that.

Comment (dczulada) [reply app/assets/javascripts/QDMPatient.js:106]: I can't say that I'm familiar with the history for that approach.  I ran a regression in Cypress and everything works as expected with this new approach.

**Review by dczulada — APPROVED** [review]

**Review by jkotanchik-SB — APPROVED** [review]

File: app/assets/javascripts/QDMPatient.js (hunk 1 above)
Comment (adongare) [reply app/assets/javascripts/QDMPatient.js:106]: thanks for verifying it!
